src/coreclr/jit/compiler.h

```diff
@@ -215,17 +215,26 @@ class LclSsaVarDsc
     // The GT_ASG node that generates the definition, or nullptr for definitions
     // of uninitialized variables.
     GenTreeOp* m_asg;
+    // Number uses of this SSA def (may be an over-estimate). Includes phi args uses.
+    unsigned m_numUses;
+    // True if there may be phi args uses of this def
+    // (false implies all uses are non-phi)
+    bool m_hasPhiUse;
+    // True if there may be uses of the def in a different block
+    bool m_hasGlobalUse;
```

Comment:
Is it intentional these fields are not under `#ifdef DEBUG`? They will increase memory consumption quite a bit.

Reply:
Yes, I plan to rely on this information during RBO.

Reply:
How much of this information will be required in release? Can it be compressed to avoid the memory regression (by stashing the bits in some block/node pointers, for example)? It does not seem necessary to increase the size by 8 bytes, 4 should suffice for the number of uses + booleans turned bitfields.

Reply:
RBO will (at least initially) just use `m_hasGlobalUse`. A possible second pass of forward sub would also check `m_numUses` though it would only want to know if the value is 0, 1, or > 1.

I wonder how hard it would be for SPMI to track jit memory usage and report that as a release-mode metric. Seems like it should be doable.

Reply:
Indeed, it would be nice to have this information in CI.

Meanwhile, quick runs CoreLib yield this (for an equivalent change):
```
./diff-mem x64
Absolute difference is: 3876656
Relative difference is: 0.185%

./diff-mem x86
Absolute difference is: 3798960
Relative difference is: 0.270%
```